src/indra_cogex/apps/query_web_app/__init__.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+
+"""An app wrapping the query module of indra_cogex."""
+import logging
+from inspect import isfunction, signature, Signature
+from typing import Callable, Tuple, Type, Iterable, Any, Dict, List, Counter, Mapping
+
+import flask
+from docstring_parser import parse
+from flask import request, jsonify, abort, Response
+from flask_restx import Api, Resource, fields
+from more_click import make_web_command
+
+from indra.statements import Evidence, Statement, Agent
+from indra_cogex.client.neo4j_client import Neo4jClient
+from indra_cogex.client import queries
+from indra_cogex.representation import Node
+
+app = flask.Flask(__name__)
+api = Api(
+    app,
+    title="INDRA CoGEx Query API",
+    description="REST API for INDRA CoGEx queries",
+)
+
+query_ns = api.namespace("CoGEx Queries", "Queries for INDRA CoGEx", path="/api/")
+client = Neo4jClient()
+
+
+logger = logging.getLogger(__name__)
+
+
+examples_dict = {
+    "tissue": ["UBERON", "UBERON:0002349"],
+    "gene": ["HGNC", "9896"],
+    "go_term": ["GO", "GO:0000978"],
+    "drug": ["CHEBI", "CHEBI:27690"],
+    "disease": ["MESH", "D007855"],
+    "trial": ["CLINICALTRIALS", "NCT00000114"],
+    "genes": [["HGNC", "1097"], ["HGNC", "6407"]],
+    "pathway": ["WIKIPATHWAYS", "WP5037"],
+    "side_effect": ["UMLS", "C3267206"],
+    "term": ["MESH", "D007855"],
+    "parent": ["MESH", "D007855"],
+    "mesh_term": ["MESH", "D015002"],
+    "pmid_term": ["PUBMED", "27890007"],
+    "include_child_terms": True,
+    # NOTE: statement hashes are too large to be int for JavaScript
+    "stmt_hash": "12198579805553967",
+    "stmt_hashes": ["12198579805553967", "30651649296901235"],
+    "cell_line": ["CCLE", "BT20_BREAST"],
+    "target": ["HGNC", "6840"],
+    "include_indirect": True,
+    "evidence_map": {},
+    "filter_medscan": True,
+}
+
+
+def parse_json(query_json: Dict[str, Any]) -> Dict[str, Any]:
+    """Parse the incoming query
+
+    Parameters
+    ----------
+    query_json :
+        The incoming query as a dictionary
+
+    Returns
+    -------
+    :
+        The parsed query
+    """
+    parsed_query = {}
+    for key, value in query_json.items():
+        if key in ('stmt_hashes', 'stmt_hash'):
+            if isinstance(value, str):
+                parsed_query[key] = int(value)
+            elif isinstance(value, list):
+                parsed_query[key] = [int(v) for v in value]
+            else:
+                raise ValueError(f"{key} must be a string or list of strings")
+        else:
+            parsed_query[key] = value
+
+    return parsed_query
+
+
+def process_result(result) -> Any:
+    # Any fundamental type
+    if isinstance(result, (int, str, bool)):
```

Comment:
float?

Reply:
Fixed in https://github.com/bgyori/indra_cogex/pull/70/commits/bfad9aba3bc0cfb785c874e02753d4955460df61